### DISCUSSION TITLE
@Scheduled execution behavior

### DISCUSSION THREAD

**Discussion by Nahuel92** [micronaut-projects/micronaut-core#11257] [General]
Hi there!

I have a question about `@Scheduled`, and I couldn't find anything about it in the docs.

I have a method annotated with this annotation, and it runs some tasks using a virtual thread executor. My question is, will Micronaut wait until all virtual threads are done before executing again the job method? Or will it execute it again no matter what, even if processing overlaps?

Thanks!

**Comment by yawkat** [comment]
unless the method waits until those tasks are complete, the scheduler will not consider the subtasks, no.

**Reply by Nahuel92** [reply]
I see. Can you give me any suggestion on how I can make a scheduler to wait until all tasks are completed before executing again? I read on another issue/discussion that one has to create a custom Scheduler, but I'm not sure on how to do so

**Reply by Nahuel92** [reply]
Please disregard, I'll make my code to run directly in the annotated method without using virtual threads
